source/02_algo/exploration_rl_en.rst

```diff
@@ -0,0 +1,241 @@
+
+The more novel a certain state\ :math:`s`\ is, the more often the corresponding agent has insufficient cognition of the state\ :math:`s`\, and the agent needs to explore this state more when interacting with the environment later \ :math:`s`\ , so this specially designed exploration mechanism confers a greater intrinsic reward on that state. How to measure the novelty of the state\ :math:`s`\? There are two main ways, one is to measure the state by counting the state in some way, and the other is to measure based on the prediction error of a certain prediction problem, so as to obtain \ **Exploration based on intrinsic reward**\ the following 2 large subclasses of algorithms: \ **Count-based intrinsic reward**\ and \ **Prediction error-based intrinsic reward**\.
+
+Count-Based Intrinsic Reward
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Count-based intrinsic reward adopts the simplest idea of measuring novelty by counting, i.e. each :math:`s` corresponds to a visit count\ :math:`N(s)`\  , the larger the value, the more times the agent has visited it before, that is, the exploration of:math:`s`  The more sufficient (or:math:`s` less novel). The exploration module gives an intrinsic reward that is inversely proportional to the state count.
+
+-  `Unifying Count-Based Exploration and Intrinsic
+   Motivation <https://arxiv.org/abs/1606.01868>`__
+   [1]_ uses a density model to approximate the frequency of state visits and proposes a novel pseudo-count derived from the density model algorithm.
+
+-  `Exploration: A Study of Count-Based Exploration for Deep
+   Reinforcement Learning <https://arxiv.org/abs/1611.04717>`__
+   [2]_ proposes to use Locality-Sensitive Hash (LSH ) to convert continuous high-dimensional state data into discrete hash codes. Thus, the statistics of the frequency of occurrence of states become feasible.
+
+However, count-based methods for measuring novelty have many obvious limitations:
+
+-  There is no simple counting method for high-dimensional continuous observation space and continuous action space.
+
+-  The visit count does not accurately measure the agent's awareness of :math:`( s,a )`.
+
+Intrinsic Reward Based on Prediction Error
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Intrinsic reward based on prediction error is  \ **to use the prediction error of a state on a prediction problem (usually a supervised learning problem) to measure novelty**\. According to the characteristics of neural network fitting datasets in supervised learning, if the agent has a larger prediction error in a certain state \ :math:`s`\, the approximate state space in the vicinity of the state \ :math:`s`\. The number of previous visits by the upper agent is small, so the state\ :math:`s`\ is more novel.
```

Comment:
Grammar mistake in "According to the characteristics of neural network fitting datasets in supervised learning, if the agent has a larger prediction error in a certain state \ :math:`s`\, the approximate state space in the vicinity of the state \ :math:`s`\."